src/lib/mdns/Advertiser.h

```diff
@@ -285,30 +285,68 @@ class CommissionAdvertisingParameters : public BaseAdvertisingParams<CommissionA
     bool mPairingInstrHasValue = false;
 };
 
-/// Handles advertising of CHIP nodes
+/**
+ * Interface for advertising CHIP DNS-SD services.
+ *
+ * A user of this interface must first initialize the advertiser using the `Init` method.
+ *
+ * Then, whenever advertised services need to be refreshed, the following sequence of events must
+ * occur:
+ * 1. Call the `ResetServices` method.
+ * 2. Call one of the `Advertise` methods for each service to be added or updated.
+ * 3. Call the `CommitServices` method to complete the update and apply all pending changes.
+ */
 class ServiceAdvertiser
 {
 public:
     virtual ~ServiceAdvertiser() {}
 
-    /// Starts the advertiser. Items 'Advertised' will become visible.
-    /// Must be called before Advertise() calls.
-    virtual CHIP_ERROR Start(chip::Inet::InetLayer * inetLayer, uint16_t port) = 0;
-
-    /// Advertises the CHIP node as an operational node
+    /**
+     * Initializes the advertiser.
+     *
+     * The method must be called before other methods of this class.
+     * If the advertiser has already been initialized, the method exits immediately with no error.
+     */
+    virtual CHIP_ERROR Init(chip::Inet::InetLayer * inetLayer, uint16_t port) = 0;
+
+    /**
+     * Shuts down the advertiser.
+     */
+    virtual void Shutdown() = 0;
+
+    /**
+     * Removes or marks all services being advertised for removal.
+     *
+     * Depending on the implementation, the method may either stop advertising existing services
+     * immediately, or mark them for removal upon the subsequent `CommitServices` method call.
+     */
+    virtual CHIP_ERROR ResetServices() = 0;
```

Comment:
I'm not sure if `ResetServices` name is the best choice. Reset for me doesn't clearly suggest what is going to happen with them, as it's like bringing back to the initial state, that is no service registered? Looking at the description maybe it could be named just `RemoveServices`?

Reply:
Yeah, you may be right. I just didn't want to make the impression that the function actually removes them immediately, but couldn't find the right word.

Reply:
So maybe `RequestRemovingServices`? That describes exactly what it does.

Reply:
I'll rename it to `RemoveServices`.

Reply:
I think neither name precisely describes what the method is meant for, so I'm choosing a shorter `RemoveServices` and let the comment describe the intent. :)